File: packages/sdk-typescript/examples/quickstart.ts
// Quickstart: drive a Driftstack session end-to-end.
//
// Run with: DRIFTSTACK_API_KEY=ds_live_... npx tsx examples/quickstart.ts

/* eslint-disable no-console */
import { Driftstack } from '@driftstack/sdk';

const apiKey = process.env.DRIFTSTACK_API_KEY;
if (!apiKey) {
  console.error('Set DRIFTSTACK_API_KEY in your environment.');
  process.exit(1);
}

const client = new Driftstack({ apiKey });

async function main(): Promise<void> {
  console.log('creating session…');
  const session = await client.sessions.create({ label: 'quickstart' });
  console.log('  →', session.id);

  console.log('navigating to https://example.com…');
  const nav = await client.sessions.navigate(session.id, {
    url: 'https://example.com',
    wait_until: 'load',
  });
  console.log(`  → status ${nav.status.toString()}, ${nav.duration_ms.toString()}ms`);

  console.log('capturing screenshot…');
  const shot = await client.sessions.capture(session.id, { kind: 'screenshot', full_page: false });
  console.log(`  → ${shot.byte_size.toString()} bytes (${shot.encoding})`);

  console.log('destroying session…');
  await client.sessions.destroy(session.id);
  console.log('  → done');
}

main().catch((err: unknown) => {
  console.error(err);
  process.exit(1);
});

File: packages/sdk-python/examples/quickstart.py
"""Quickstart: create a session, navigate, capture, destroy.

Run::

    DRIFTSTACK_API_KEY=ds_live_… python examples/quickstart.py

Set ``DRIFTSTACK_BASE_URL`` if you're not hitting production
(``https://api.driftstack.dev``).
"""

from __future__ import annotations

import os
import sys

from driftstack import Driftstack


def main() -> int:
    api_key = os.environ.get("DRIFTSTACK_API_KEY")
    if not api_key:
        print("DRIFTSTACK_API_KEY environment variable is required", file=sys.stderr)
        return 1

    base_url = os.environ.get("DRIFTSTACK_BASE_URL", "https://api.driftstack.dev")

    with Driftstack(api_key=api_key, base_url=base_url) as client:
        # Create a session and navigate to a page.
        session = client.sessions.create({"label": "quickstart"})
        print(f"created session {session.id}")

        client.sessions.navigate(str(session.id), {"url": "https://example.com/"})
        print("navigated")

        # Capture a screenshot. The response carries the bytes inline
        # (base64) so this works without any external blob storage.
        capture = client.sessions.capture(str(session.id), {"kind": "screenshot"})
        print(f"captured screenshot ({capture.byte_size} bytes)")

        # Destroying a session is idempotent — safe to call twice.
        client.sessions.destroy(str(session.id))
        print(f"destroyed session {session.id}")

    return 0


if __name__ == "__main__":
    sys.exit(main())
